src/App.tsx

```diff
@@ -8,7 +8,57 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+
+interface TodosFilterOptions {
+  todos: Todo[];
+  query: string;
+  filterBy: FilterType;
+}
+
+const getVisibleTodos = ({ todos, query, filterBy }: TodosFilterOptions): Todo[] => {
+  let visibleTodos = [...todos];
+
+  if (query) {
+    const normalizedQuery = query.trim().toLowerCase();
+
+    visibleTodos = visibleTodos.filter((todo) => todo.title.toLowerCase().includes(normalizedQuery));
+  }
+
+  if (filterBy) {
+    visibleTodos = visibleTodos.filter((todo) => {
+      switch (filterBy) {
+        case FilterType.Active:
+          return !todo.completed;
+        case FilterType.Completed:
+          return todo.completed;
+        default:
+          return FilterType.All;
+      }
+    });
+  }
+
+  return visibleTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/96409563/eb1dc5be-4641-47ef-a7eb-d6b8eaeba236)